src/server/httpServer.ts

```diff
@@ -167,7 +170,8 @@ export class HttpServer extends Disposable {
 		}
 
 		let looseFile = false;
-		let absoluteReadPath = path.join(basePath ?? '', unescape(URLPathName));
+		let absoluteReadPath = path.join(basePath ?? '', decodeURI(URLPathName));
+		URLPathName = decodeURI(URLPathName)

```

Comment:
nit: you can put this line above line 173 and then you don't need to use `decodeURI` within the `path.join` args.

Reply:
oh!! I didn't notice